### ISSUE TITLE
Crash while processing Raspverry Pi photos

### ISSUE THREAD

**Issue by gennyble** [gennyble/libraw-colordata#1]
Is this our fault, libraw-rs's or libraw's?

**Comment by gennyble** [comment]
I thought this might be because of the unsafe code at the start of `libraw-rs::Colordata::new`. I commented it out, made the profile a `vec![]`, and the RaspberryPi folder still crashed it.

**Comment by gennyble** [comment]
It's libraw-rs, libraw-sys, or libraw. I wrote a ltitle MVP to parse RaspberryPi.

<details>
<summary>RaspberryPi Folder Structure</summary>
<pre><code>RP_imx219
- Pi_V2.1_picamera.raw
- Pi_V2.1.raw
RP_imx477
- pi.dng
- pi.raw
- pi_hq_raspistill.raw
- pi_hq_picamera.raw
- pi_hq_raspistill.dng
- pi_hq_picamera.dng
RP_ov5647
- Pi_V1.3.raw
- Pi_V1.3_picamera.raw
</code></pre>
</details>

<details>
<summary>Output with the directories in a few different orders</summary>
When imx219 is first I get
<pre><code>fish: Job 1, 'cargo run' terminated by signal SIGSEGV (Address boundary error)
</code></pre>

When imx477 is first I get
<pre><code>Parsed RaspberryPi/RP_imx477/pi.dng
Parse failed RaspberryPi/RP_imx477/pi.raw
	libraw error: -2
Parse failed RaspberryPi/RP_imx477/pi_hq_raspistill.raw
	libraw error: -2
Parse failed RaspberryPi/RP_imx477/pi_hq_picamera.raw
	libraw error: -2
Parsed RaspberryPi/RP_imx477/pi_hq_raspistill.dng
Parsed RaspberryPi/RP_imx477/pi_hq_picamera.dng
Parsed RaspberryPi/RP_imx219/Pi_V2.1.raw
free(): corrupted unsorted chunks
fish: Job 1, 'cargo run' terminated by signal SIGABRT (Abort)
</code></pre>

With the folders reversed I get
<pre><code>Parse failed RaspberryPi/RP_ov5647/Pi_V1.3.raw
	libraw error: -2
Parse failed RaspberryPi/RP_ov5647/Pi_V1.3_picamera.raw
	libraw error: -2
Parsed RaspberryPi/RP_imx477/pi.dng
Parse failed RaspberryPi/RP_imx477/pi.raw
	libraw error: -2
Parse failed RaspberryPi/RP_imx477/pi_hq_raspistill.raw
	libraw error: -2
Parse failed RaspberryPi/RP_imx477/pi_hq_picamera.raw
	libraw error: -2
Parsed RaspberryPi/RP_imx477/pi_hq_raspistill.dng
Parsed RaspberryPi/RP_imx477/pi_hq_picamera.dng
Parsed RaspberryPi/RP_imx219/Pi_V2.1.raw
free(): corrupted unsorted chunks
fish: Job 1, 'cargo run' terminated by signal SIGABRT (Abort)
</code></pre>
</details>

**TL;DR**
It seems like the crash is only with the `RP_imx219` model of camera. If you parse other Raspberry Pi camera models before it you seem to get different results. Notably, parsing the imx219 first gives a SIGSEGV but parsing it after you've parsed another model you get `free(): corrupted unsorted chunks` and then `SIGABRT`.

I can't see how this would be an issue in libraw-rs, but I'm not super familiar with how it interacts with libraw-sys while parsing. I'm pretty sure this is a libraw issue, but I'll try and use libraw-sys myself first.
